packages/malloy/src/lang/parse-tree-walkers/find-table-path-walker.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files
+ * (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import {MalloyParserListener} from '../lib/Malloy/MalloyParserListener';
+import {getId, getPlainString} from '../parse-utils';
+import {MalloyTranslation} from '../parse-malloy';
+import {CommonTokenStream} from 'antlr4ts';
+import {DocumentRange} from '../../model/malloy_types';
+import * as parser from '../lib/Malloy/MalloyParser';
+import {MalloyParseInfo} from '../malloy-parse-info';
+import {ParseTreeWalker} from 'antlr4ts/tree/ParseTreeWalker';
+
+export interface PathInfo {
+  connId: string;
+  tablePath: string;
+  range: DocumentRange;
+}
+
+class FindTablePathWalker implements MalloyParserListener {
+  pathInfos: PathInfo[] = [];
+  constructor(
+    readonly translator: MalloyTranslation,
+    readonly tokens: CommonTokenStream
+  ) {}
+
+  enterTableMethod(pcx: parser.TableMethodContext): void {
+    const connId = getId(pcx.connectionId());
+    const tablePath = getPlainString(pcx.tablePath());
```

Comment:
It seems like the functionality of `getPlainString` is not super clear to the caller. It looks like it returns the non-interpolated parts of a string, which might be strange behavior in this particular context.

If you write `duckdb.table("""foo.bar.baz%{foo}""")` you will get a parser error, but then we will attempt to show you table path info for `"foo.bar.baz"` which seems wrong.

Maybe you should rework this to just exclude table calls that have interpolations.

Reply:
Done.